misc/userscripts/qute-pass

```diff
@@ -0,0 +1,142 @@
+#
+# Dependencies: tldextract (Python 3 module), pass
+# For issues and feedback please use: https://github.com/cryzed/qutebrowser-userscripts
+#
+# Chris Braun (cryzed), 2017
+
+import argparse
+import enum
+import fnmatch
+import functools
+import os
+import re
+import shlex
+import subprocess
+import sys
+
+import tldextract
+
+PASSWORD_STORE_PATH = os.path.expanduser('~/.password-store')
+DMENU_INVOCATION = 'rofi -dmenu'
+ENCODING = 'UTF-8'
+
+argument_parser = argparse.ArgumentParser()
+argument_parser.add_argument('url', nargs='?', default=os.getenv('QUTE_URL'))
+argument_parser.add_argument('--password-store', '-p', default=PASSWORD_STORE_PATH,
+                             help='Path to your pass password-store')
+argument_parser.add_argument('--username-pattern', '-u', default=r'.*/(.+)',
+                             help='Regular expression that matches the username')
+argument_parser.add_argument('--username-target', '-U', choices=['path', 'secret'], default='path',
+                             help='The target for the username regular expression')
+argument_parser.add_argument('--password-pattern', '-P', default=r'(.*)',
+                             help='Regular expression that matches the password')
+argument_parser.add_argument('--dmenu-invocation', '-d', default='rofi -dmenu',
+                             help='Invocation used to execute a dmenu-provider')
+argument_parser.add_argument('--no-insert-mode', '-n', dest='insert_mode', action='store_false',
+                             help="Don't automatically enter insert mode")
+group = argument_parser.add_mutually_exclusive_group()
+group.add_argument('--username-only', '-uo', action='store_true', help='Only insert username')
+group.add_argument('--password-only', '-po', action='store_true', help='Only insert password')
+
+stderr = functools.partial(print, file=sys.stderr)
+
+
+class ExitCodes(enum.IntEnum):
+    SUCCESS = 0
+    COULD_NOT_DETERMINE_URL = 1
+    COULD_NOT_DETERMINE_DOMAIN = 2
+    NO_PASS_CANDIDATES = 3
+    COULD_NOT_MATCH_USERNAME = 4
+    COULD_NOT_MATCH_PASSWORD = 5
+
+
+def qute_command(command):
+    with open(os.environ['QUTE_FIFO'], 'w') as fifo:
+        fifo.write(command + '\n')
+        fifo.flush()
+
+
+def find_pass_candidates(domain, password_store_path=PASSWORD_STORE_PATH):
+    candidates = []
+    for path, directories, file_names in os.walk(password_store_path):
+        if directories or domain not in path.split(os.path.sep):
+            continue
+
+        # Strip password store path prefix to get the relative pass path
+        pass_path = path[len(password_store_path) + 1:]
+        secrets = fnmatch.filter(file_names, '*.gpg')
+        candidates.extend(os.path.join(pass_path, os.path.splitext(secret)[0]) for secret in secrets)
+    return candidates
+
+
+def pass_(path):
+    process = subprocess.run(['pass', path], stdout=subprocess.PIPE, encoding=ENCODING)
```

Comment:
`encoding` was added in Python 3.6, so you might prefer to not do this and do `process.stdout.decode('utf-8').strip()` instead.

Reply:
Good catch, backwards compatibility with earlier Python versions is something I changed before submitting the pull request.